ammonia_test_script.py

```diff
@@ -0,0 +1,71 @@
+#####################################################################
+# This script tests the ammonia fitter                              #
+# with fake data produced by ammonia_model.n_modelfunc              #
+# using the xarr from G031.947+00.076_nh3_11_Tastar.fits [for now]  #
+#####################################################################
+
+from pyspeckit.spectrum.models import ammonia
+from pyspeckit import Spectrum
+from pyspeckit.spectrum.units import SpectroscopicAxis
+from astropy import units as u
+import numpy as np
+import matplotlib.pyplot as plt
+
+passes = []
+fails = []
+passdiff = []
+faildiff = []
+results = {'pass': passes, 'pass difference': passdiff,
+           'fail': fails, 'fail difference': faildiff}
+
+def generate_guesses(starting_params, parameter_error_amplitude):
+    guesses = [starting_params[0] + (np.random.randn() * parameter_error_amplitude[0]), 
+               starting_params[1] + (np.random.randn() * parameter_error_amplitude[1]),
+               starting_params[2] + (np.random.randn() * parameter_error_amplitude[2]),
+               starting_params[3] + (np.random.randn() * parameter_error_amplitude[3]),
+               starting_params[4] + (np.random.randn() * parameter_error_amplitude[4]),
+               (np.random.randn() * parameter_error_amplitude[5])]
+    return guesses    
+
+parameter_error_amplitude = [0.2, 0.5, 1e2, 1., 1., 1.]
```

Comment:
I will adjust these because right now they the fit is failing 99/100 times,  the 3rd parameter (N) is the problem the difference sometimes spikes up to e28